internal/ingest/ingest.go

```diff
@@ -70,6 +73,14 @@ func NewIngester(cfg config.Ingest, h host.Host, idxr *indexer.Engine, reg *regi
 			efsb.Insert("PreviousID", ssb.ExploreRecursiveEdge())
 		}).Node()
 
+	// Construct the selector used when syncing entries of an advertisement with the configured
+	// recursion limit.
+	entSel := ssb.ExploreRecursive(cfg.EntriesRecursionLimit(),
+		ssb.ExploreFields(func(efsb builder.ExploreFieldsSpecBuilder) {
+			efsb.Insert("Next", ssb.ExploreRecursiveEdge())    // Next field in EntryChunk
+			efsb.Insert("Entries", ssb.ExploreRecursiveEdge()) // Entries field in advertisement
```

Comment:
Why do we need to expore the `Entries` field? Aren't we already saying we are in an entrychunk, so we wouldn't expect to run into an advertisement.

Reply:
Only added so that this works even if given CID of an advertisement. Happy to remove if you think that's better.

Reply:
Yes please. I think it's a bit simpler to reason about and that way it fails if someone sends us an entry chunk that links to an advertisement (it would be weird if that worked!).